auth_mellon_util.c

```diff
@@ -574,16 +574,25 @@ int am_read_post_data(request_rec *r, char **data, apr_size_t *length)
     bytes_left = len;
 
     while (bytes_left > 0) {
-        /* Read data from the client. Returns 0 on EOF or error, the
-         * number of bytes otherwise.
+        /* Read data from the client. Returns 0 on EOF and -1 on
+         * error, the number of bytes otherwise.
          */
         read_length = ap_get_client_block(r, &(*data)[bytes_read],
                                           bytes_left);
         if (read_length == 0) {
+            // got the EOF
+            break;
```

Comment:
Now that getting an "early" EOF (before we get the expected number of bytes) isn't an error, I think the code needs to adjust a couple of things:

* Add a null terminator (like is done at [line 568](https://github.com/UNINETT/mod_auth_mellon/pull/71/files#diff-e746ce33fcd1ce691925112f3a4be984R568)).
* Update `length` with the actual number of bytes read (as on [line 564](https://github.com/UNINETT/mod_auth_mellon/pull/71/files#diff-e746ce33fcd1ce691925112f3a4be984R564)).